src/Components/Slideshow/Slideshow.js

```diff
@@ -10,185 +10,140 @@ import './Slideshow.css'
-    if (this.animating) return
-    const nextIndex =
-      this.state.activeIndex === items.length - 1
-        ? 0
-        : this.state.activeIndex + 1
-    this.setState({ activeIndex: nextIndex })
+  function handlePrevious () {
+    if (animating) return
+    const nextIndex = currentSlide === 0 ? items.length - 1 : currentSlide - 1
+    setCurrentSlide(nextIndex)
   }
 
-  handlePrevious () {
-    if (this.animating) return
-    const nextIndex =
-      this.state.activeIndex === 0
-        ? items.length - 1
-        : this.state.activeIndex - 1
-    this.setState({ activeIndex: nextIndex })
+  function handleGoToIndex (newIndex) {
+    if (animating) return
+    setCurrentSlide(newIndex)
   }
 
-  handleGoToIndex (newIndex) {
-    if (this.animating) return
-    this.setState({ activeIndex: newIndex })
-  }
-
-  render () {
-    const { activeIndex } = this.state
-
-    const slides = items.map((item, index) => {
-      return (
-        <CarouselItem
-          id={'caption-' + index}
-          className='slideshow text-center'
-          tag='div'
-          key={item.id}
-          onExiting={this.handleOnExiting}
-          onExited={this.handleOnExited}
-        >
-          <img
-            src={item.src}
-            alt={item.altText}
-            className='carousel-images img-fluid'
-            style={{ top: item.top }}
-          />
-          <CarouselCaption
-            captionText={item.caption}
-            captionHeader={item.header}
-          />
-        </CarouselItem>
-      )
-    })
-
-    return (
-      <div id='clicker'>
-        <style>
-          {`.slideshow {
+  return (
+    <div id='clicker'>
+      <style>
+        {`.slideshow {
                 align-items: center;
                 height: 94vh;
               }`}
-        </style>
-        <Carousel
-          activeIndex={activeIndex}
-          next={this.onNext}
-          previous={this.onPrevious}
-        >
-          <CarouselIndicators
-            items={items}
-            activeIndex={activeIndex}
-            onClickHandler={this.handleGoToIndex}
-          />
-          {slides}
-          <CarouselControl
-            direction='prev'
-            directionText='Previous'
-            onClickHandler={this.handlePrevious}
-          />
-          <CarouselControl
-            direction='next'
-            directionText='Next'
-            onClickHandler={this.handleNext}
-          />
-        </Carousel>
-      </div>
-    )
-  }
+      </style>
+      <Carousel activeIndex={currentSlide} next={onNext} previous={onPrevious}>
+        <CarouselIndicators
+          items={items}
+          activeIndex={currentSlide}
+          onClickHandler={handleGoToIndex}
+        />
+        {items.map((item, index) => {
+          return (
+            <CarouselItem
+              id={'caption-' + index}
+              className='slideshow text-center'
+              tag='div'
+              key={index}
+              onExiting={handleOnExiting}
```

Comment:
```suggestion
              onExiting={() => setAnimating(false)}
```

Reply:
I believe this would be onExiting-{() => setAnimating(true)}  since previously handOnExiting was set to true. Is that correct?